Read nidm packs and overwrite existing folder

pwilliams1 opened the pull request:
This PR is work-in-progress and includes:
- Begin adding .nidm.zip option
- Add overwrite html folder option (command line argument)

pwilliams1 committed:
Modify testError.py to run viewer against all turtle files required

pwilliams1 committed:
Edit function queryStatisticType to include spm_PartialConjuctionInference

pwilliams1 committed:
Begin adding option to overwite html folder (command line argument)

pwilliams1 committed:
Add command line argument for overwriting html folder

pwilliams1 committed:
Add comments

pwilliams1 committed:
Set overwite argument to True (for html folder)

pwilliams1 committed:
Begin adding option to handle nidm zipfile

pwilliams1 committed:
Add code to extract nidm zip file and find the turtle file

pwilliams1 committed:
Overwrite option not working

pwilliams1 committed:
Remove line checking for specific turtle files

pwilliams1 committed:
Add comments edit zip file option

pwilliams1 committed:
Allow html folder to be overwritten during tests and normal usage - Backup option added in case of future errors

pwilliams1 committed:
Add code to extract files from nidm zip file, and to run the viewer on the turtle file inside the zip

pwilliams1 committed:
Add folder overwrite into test_fslViewer.py

pwilliams1 committed:
Add if-else branch in main function to handle zip and turtle file cases

pwilliams1 committed:
Modify main function to place html files directly into extracted zip file folder, to solve the problem of the html page not displaying the images

pwilliams1 committed:
Fix destinationFolder variable not being assigned

pwilliams1 committed:
Begin adding option to overwrite html folder for nidm zip files

pwilliams1 committed:
Add code to delete html folder if user specifies to

pwilliams1 committed:
Add comments and remove print statements

cmaumet commented on nidmviewerfsl/viewer.py:
```diff
@@ -529,26 +531,88 @@ def createOutputDirectory(outputFolder): #Attempts to create folder for HTML fil
+				zip.extractall(htmlFolder) #Extract zip file to destination folder
+				turtleFile = glob.glob(os.path.join(htmlFolder, "*.ttl"))
+				print(turtleFile)
+				g.parse(turtleFile[0], format = rdflib.util.guess_format(turtleFile[0]))
```
Could we use "turtle" as format:

```
g.parse(turtleFile[0], format = turtle)
```

cmaumet reviewed (changes requested):
Hi @pwilliams1! 

To ensure that the reading of nidm packs is working as expected, can you update the tests to work with the NIDM packs rather than directly with the turtle?

pwilliams1 committed:
Begin editing tests to handle nidm zip files

pwilliams1 committed:
Modify testError.py to download nidm zip files instead of turtle files

pwilliams1 committed:
Modify testError.py to run viewer on zip files

pwilliams1 committed:
Add comments

pwilliams1 committed:
Modify test_fslViewer.py to work with nidm zip files instead of ttl files

pwilliams1 committed:
Modify viewer to use turtle as format (vs rdflib.util.guess)

Commit:
fix: check if NIDM packs are local (instead of turtle)

Commit:
If overwrite is set to True then we should not ask user input

This can happen when calling testError.py which is an issue as the
tests should run without user interaction.

Commit:
Git should ignore test data

Commit:
Update nidmviewerfsl help to say "pack" instead of "turtle"

Commit:
Remove .nidm.zip from test folder name

cmaumet reviewed (approved):
Hi @pwilliams1,

Thank you for this update. I have reviewed your code and I think it is ready for merge. 

I have added a couple of small edits to your pull request:
 - b258269: if `overWrite` is set to `True` then the viewer should not ask for user input before overwriting.
 - fbc81ec: check that the NIDM-Results pack rather than the turtle file is present to decide whether test data should be downloaded or not.
 - 8b2aa8a: Add the `Test/data` folder to the `.gitignore` file so that Git do not try to version-control them.
 - c1a6227: Remove ".nidm.zip" from the output folder names for the tests

I also have a couple of suggestions for improving the viewer (cf #7 and #8). We can look at these later on in separate pull requests.

Commit:
fix: out folder is now suffixes with _test

cmaumet commented:
Hi @pwilliams1,

I have just merged your pull request. To update the code on your local machine, please follow those steps:
 1. Make sure you are in a clean state (`git status` returns `nothing to commit, working directory clean`) if not use `git stash save`
 2. Go in the  `master` branch:
 `git checkout master`
 3. Update your local copy of the `master` branch according to what is on GitHub:
 `git pull upstream master`
 4. Try calling `git log` and check the last commit (before the merge commit) is e9bbacf.
 5. If (and only if) you had any pending changes in 1. use `git stash pop` to reapply those changes.
